js/DropZone/DropZoneComponent.js

```diff
@@ -0,0 +1,564 @@
+            this.instanceManager.addInstance(node, this.optionsManager);
+        });
+
+        // iterate over our DropZoneInstanceManagerInstances to process any input nodes & update help state
+        this.instanceManager.getInstance().forEach(instance => {
+            if (instance.input) {
+                this.processInputNode(instance.input, instance.id, instance.options.showInputNode);
+
+                if (instance.browse) {
+                    this.processBrowseNode(instance.browse, instance.input);
+                }
+            }
+        });
+
+        // bind this here, so we have a ref
+        this.removeFile = this.removeFile.bind(this);
+    }
+
+    /**
+     * If an input node ID has been passed in, we'll take care of linking the DropZone
+     * instance to this node
+     * @param {Element} input
+     * @param {number} id
+     * @param {boolean} show
+     */
+    processInputNode (input, id, show) {
+        // for now these handlers can stay anonymous, the likely hood of wanting to disable a DropZone
+        // at some point in a session seems unlikely. famous. last. words.
+        input.addEventListener('change', () => {
+            const files = input.files;
+
+            // we'll add a persist property to our file objects, this can be used to
+            // persist the front-end validation, which is essential when using an
+            // associated file input
+            if (input.value) {
+                this.instanceManager.addFiles(files, id, { persist: true });
+            }
+
+            // reset input node value, this will ensure our change event
+            // fires each time we use the browse files functionality - even
+            // if we try to add an identical value
+            input.value = '';
+        });
+
+        // visually hide input - this should ideally be done in the CSS also to prevent a
+        // flash on load (with some consideration for non javascript users)
+        if (!show) {
+            input.style.display = 'none';
+        }
+    }
+
+    /**
+     * Adds the browse files functionality to our DropZone
+     * @param {Element} browse
+     * @param {Element} input
+     */
+    processBrowseNode (browse, input) {
+        browse.addEventListener('click', event => {
+            event.preventDefault();
+            // trigger a synthetic click on the input node which will activate the
+            // native Browse Files interaction
+            input.click();
+        });
+    }
+
+    /**
+     * Update DropZone info node innerHTML
+     * @param {number} id
+     * @param {String} string
+     */
+    updateInfoState (id, string) {
+        const { info, input, node } = this.instanceManager.getInstance(id);
+        const options = this.optionsManager.getInstanceOptions(id);
+        let browse;
+
+        if (info) {
+            info.innerHTML = string;
+            browse = node.querySelector(`.${options.nodeClasses.browse}`);
```

Comment:
This is a good call, likely another "used to be a legit let but then got refactored to within an inch of it's life" scenarios 😄